### PR TITLE
[docs] [Drivers and ORMs] updated csharp pages

### PR DESCRIPTION
**Pull request by aishwarya24**
Related to PR https://github.com/yugabyte/yugabyte-db/pull/12200

- Updated csharp pages with similar changes done to the Java pages. 
https://deploy-preview-12523--infallible-bardeen-164bc9.netlify.app/preview/drivers-orms/csharp/

@netlify  /preview/drivers-orms/csharp

### CODE REVIEW

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 1]
```diff
@@ -72,9 +70,8 @@ NpgsqlConnection conn = new NpgsqlConnection(yburl)
...
+Set up the driver properties to configure the credentials and SSL Certificates for connecting to your cluster.
+The .NET Npgsql driver validates certificates differently from other PostgreSQL drivers. When you specify SSL mode `require`, the driver verifies the certificate by default (like the `verify-ca` or `verify-full` modes), and fails for self-signed certificates (like YugabyteDB's). You can override this by specifying `Trust Server Certificate = true`, in which case it bypasses walking the certificate chain to validate trust, and hence works like other drivers' `require` mode. In this case, the Root-CA certificate is not required to be configured.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:74]: extra line break or new paragraph?

For a local install, do I even use a certificate? Where would I get one? YB Managed uses certificates and normally works with verify-full; is it something about this driver that keeps it from working? ie, require + Trust Server Certificate = true is not the same as verify-full?
Following the Use C# with SSL link, table lists verify-full and verify-ca, but then the note says they aren't supported.

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 2]
```diff
@@ -90,14 +87,16 @@ connStringBuilder.TrustServerCertificate = true;
 CRUD(connStringBuilder.ConnectionString);
 ```
 
-| Parameter | Description | Default |
+| Npgsql Parameter | Description | Default |
 | :---------- | :---------- | :------ |
 | sslmode  | SSL Mode | require
 | TrustServerCertificate |  Trust the server certificate configured on the YugabyteDB cluster | true
 
+If you created a cluster on [YugabyteDB Managed](https://www.yugabyte.com/managed/), use the cluster credentials and [download the SSL Root certificate](/preview/yugabyte-cloud/cloud-connect/connect-applications/). Refer to [Use C# with SSL](../../../quick-start/build-apps/csharp/ysql/#use-c-with-ssl) for the default and supported modes.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:95]: Except Use C# with SSL says disable is supported - which it is by the driver - but is not by YB Managed...

Also, on the Use C# with SSL page there is a double colon typo

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 3]
```diff
@@ -90,14 +87,16 @@ connStringBuilder.TrustServerCertificate = true;
...
+
 ### Step 3: Query the YugabyteDB cluster from your application
 
-Next, copy the following sample code to the `Program.cs` file to set up YugbyteDB tables and query the table contents from the C# client. Ensure you replace the connection string `yburl` with the credentials of your cluster and SSL certificates if required.
+Copy the following code to the `Program.cs` file to set up YugbyteDB tables and query the table contents from the C# client. Replace the connection string `yburl` with the credentials of your cluster, and SSL certificates if required.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:99]: No "yburl" in code that follows

**Review by ddhodge — COMMENTED** [review]
Some questions

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 4]
```diff
@@ -90,14 +87,16 @@ connStringBuilder.TrustServerCertificate = true;
 CRUD(connStringBuilder.ConnectionString);
 ```
 
-| Parameter | Description | Default |
+| Npgsql Parameter | Description | Default |
 | :---------- | :---------- | :------ |
 | sslmode  | SSL Mode | require
 | TrustServerCertificate |  Trust the server certificate configured on the YugabyteDB cluster | true
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:93]: Above it says "You can override this by specifying Trust Server Certificate = true" - so is true really the default? If require fails without TrustServerCertificate doesn't that mean it is false by default? Also, above it is Trust Server Certificate, here it is TrustServerCertificate

File: docs/content/preview/drivers-orms/csharp/_index.md [hunk 5]
```diff
@@ -13,30 +12,38 @@ menu:
...
+To develop C# applications for YugabyteDB, you need the following:
 
 - **.NET SDK**\
   Install .NET SDK 3.1 or later. To download it for your supported OS, visit [Download .NET](https://dotnet.microsoft.com/en-us/download).
 
-- **Visual Studio (optional)**\
-  For ease-of-use, use an integrated development environment (IDE) such as Visual Studio. To download and install Visual Studio, visit the [Visual Studio Downloads](https://visualstudio.microsoft.com/downloads/) page.
-
 - **Create a C# project**
+   For ease-of-use, use an integrated development environment (IDE) such as Visual Studio. To download and install Visual Studio, visit the [Visual Studio Downloads](https://visualstudio.microsoft.com/downloads/) page.
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/drivers-orms/csharp/_index.md:46]: @ddhodge missed the slash while saving the file. Thanks! will make the change.

File: docs/content/preview/reference/drivers/csharp/postgres-npgsql-reference.md [hunk 6]
```diff
@@ -25,39 +25,36 @@ showAsideToc: true
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/postgres-npgsql-reference.md:25]: Everywhere - headings are sentence case

**Review by ddhodge — COMMENTED** [review]
Misc comments

File: docs/content/preview/quick-start/build-apps/csharp/ysql.md [hunk 7]
```diff
@@ -51,7 +51,7 @@ This tutorial assumes that you have:
 
 {{< warning title="Warning" >}}
 
-On every new connection the NpgSQL driver also makes [extra system table queries to map types](https://github.com/npgsql/npgsql/issues/1486), which adds significant overhead. To turn off this behavior, set the following option in your connection string builder:
+On every new connection, the Npgsql driver also makes [extra system table queries to map types](https://github.com/npgsql/npgsql/issues/1486), which adds significant overhead. To turn off this behavior, set the following option in your connection string builder:
 
 ```csharp
 connStringBuilder.ServerCompatibilityMode = ServerCompatibilityMode.NoTypeLoading;
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/quick-start/build-apps/csharp/ysql.md:57]: I understand the inconsistency across the files. I'll rename and include the string builder.

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 8]
```diff
@@ -51,53 +50,57 @@ dotnet add package Npgsql
...
+    connStringBuilder.Password = "xxxxxx";
+    connStringBuilder.Database = "yugabyte";
+    CRUD(connStringBuilder.ConnectionString);
 ```
 
-| Parameter | Description | Default |
-| :---------- | :---------- | :------ |
-| sslmode  | SSL Mode | require
-| TrustServerCertificate |  Trust the server certificate configured on the YugabyteDB cluster | true
+If you created a cluster on [YugabyteDB Managed](https://www.yugabyte.com/managed/), use the cluster credentials and [download the SSL Root certificate](../../../yugabyte-cloud/cloud-connect/connect-applications/).
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:97]: [YugabyteDB Managed](https://www.yugabyte.com/managed/) clusters require SSL. Refer to [Connect applications](link) for instructions on how to obtain the cluster connection parameters and download the CA certificate.  
link to https://docs.yugabyte.com/preview/yugabyte-cloud/cloud-connect/connect-applications/

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 9]
```diff
@@ -51,53 +50,57 @@ dotnet add package Npgsql
...
+
+Refer to [Configure SSL/TLS](../../../reference/drivers/csharp/postgres-npgsql-reference/#configure-ssl-tls) for more information on default and supported modes with examples for setting up your connection strings when using SSL.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:99]:
```suggestion
Refer to [Configure SSL/TLS](../../../reference/drivers/csharp/postgres-npgsql-reference/#configure-ssl-tls) for more information on Npgsql default and supported SSL modes, and examples for setting up your connection strings when using SSL.
```

File: docs/content/preview/reference/drivers/csharp/postgres-npgsql-reference.md [hunk 10]
```diff
@@ -25,61 +25,56 @@ showAsideToc: true
...
+### Connect to YugabyteDB database
 
-After setting up the dependenices, we implement the C# client application that uses the Npgsql driver to connect to your YugabyteDB cluster and run query on the sample data.
+After setting up the dependencies, implement the C# client application that uses the Npgsql driver to connect to your YugabyteDB cluster and run a query on the sample data.
 
-We will import Npgsql and use the `NpgsqlConnection` class for getting connection object for the YugabyteDB Database which can be used for performing DDLs and DMLs against the database.
+Import Npgsql and use the `NpgsqlConnection` class to create the connection object to perform DDLs and DMLs against the database.
 
-Example URL for connecting to YugabyteDB can be seen below.
+Example connection string parameter (connStringBuilder) for connecting to YugabyteDB is as follows:
```
Comment (ddhodge) [inline docs/content/preview/reference/drivers/csharp/postgres-npgsql-reference.md:59]: I'd drop in the changes from the same section on the drivers page - table first, then example string

**Review by ddhodge — COMMENTED** [review]
Some comments

File: docs/content/preview/reference/drivers/csharp/postgres-npgsql-reference.md (hunk 10 above)
Comment (aishwarya24) [reply docs/content/preview/reference/drivers/csharp/postgres-npgsql-reference.md:59]: got it , re-ordered the tables on both the pages.

File: docs/content/preview/drivers-orms/csharp/postgres-npgsql.md [hunk 11]
```diff
@@ -25,22 +23,23 @@ showAsideToc: true
...
 ## CRUD operations with PostgreSQL Npgsql driver
 
-Learn how to establish a connection to YugabyteDB database and begin basic CRUD operations using the steps in the [Build an application](../../../quick-start/build-apps/csharp/ysql) page under the Quick start section.
+Learn how to establish a connection to YugabyteDB database and begin basic CRUD operations using the steps on the [Build an application](../../../quick-start/build-apps/csharp/ysql) page under the Quick start section.
 
 The following section breaks down the quick start example to demonstrate how to perform common tasks required for C# application development using the Npgsql driver.
 
 After completing these steps, you should have a working C# application that uses the Npgsql driver to connect to your cluster, set up tables, run a query, and print out results.
 
+The quick start example requires you to run the steps from the [Prerequisites](../../../quick-start/build-apps/csharp/ysql/#prerequisites) section before proceeding with building the C# application.
```
Comment (ddhodge) [inline docs/content/preview/drivers-orms/csharp/postgres-npgsql.md:36]:
```suggestion
```

**Review by ddhodge — APPROVED** [review]
Minor stuff

**Pull request merged**